plugins/modules/proxmox.py

```diff
@@ -695,20 +695,21 @@ def main():
     if state == 'present' and clone is None:
         try:
             if proxmox.get_vm(vmid, ignore_missing=True) and not module.params['force']:
-                module.exit_json(changed=False, msg="VM with vmid = %s is already exists" % vmid)
+                module.exit_json(changed=False, vmid=vmid, taskid=taskid, msg="VM with vmid = %s is already exists" % vmid)
```

Comment:
The sanity tests complain that `taskid` is used before assignment in this case.

Reply:
Oh, yup. It's definitely not getting assigned at all within that function.

It looks like grabbing the taskid from a handful of the functions is a greater task (pun not intended) than simply returning it. In fact, most of the functions in this file could stand to be cleaned up and return more structured information.

I removed the `taskid` value from most of the places where I had added it, and I don't see any more use-before-assignment issues in my own linter. I didn't run the sanity checks on my system since the CI will, but I think it should pass now. I only left it in where I'm certain that the function is assigning it properly.

---
I'm not sure how up to the task I am or not, and it's definitely a larger scope than this PR should be (originally I was just trying to return the vmid), but I think that a handful of the functions in this file could stand to be cleaned up. I'll open an issue about refactoring this file a little bit with some more specifics of what I have in mind. 

Thank you!